### PR TITLE
Add aws_s3 integration tests for multipart GET operations

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Test that multipart files are not re-downloaded if the local dest and remote object's checksums match. This feature was added in f20af4b9097edc4ff3136f93c74049e7a43cf6be.

##### ISSUE TYPE
 - Test Pull Request

##### COMPONENT NAME
test/integration/targets/aws_s3/tasks/main.yml

##### ANSIBLE VERSION
```
2.5.0
```

### CODE REVIEW

File: test/integration/targets/aws_s3/tasks/main.yml [hunk 1]
```diff
@@ -252,3 +252,76 @@
     that:
       - result.changed == False
 # ============================================================
+- name: create a tempfile for multipart test
+  tempfile:
+  register: tmp1
+
+- name: make tempfile 4 GB
+  command:
+    _raw_params: "dd if=/dev/zero of={{ tmp1.path }} bs=1m count=4096"
```
Comment (s-hertel) [inline test/integration/targets/aws_s3/tasks/main.yml]: Works on mac os x. This line needs to be fixed.

Comment (ryansb) [reply test/integration/targets/aws_s3/tasks/main.yml]: This also works on linux if you capitalize the M in `bs=1M`

File: test/integration/targets/aws_s3/tasks/main.yml [hunk 2]
```diff
@@ -252,3 +252,76 @@
...
+  aws_s3:
+    bucket: "{{ bucket_name }}"
+    mode: get
+    dest: /tmp/multipart_download.txt
+    object: multipart.txt
+    overwrite: different
+    security_token: "{{security_token}}"
+    aws_access_key: "{{ ec2_access_key }}"
+    aws_secret_key: "{{ ec2_secret_key }}"
+  register: result
```
Comment (ryansb) [inline test/integration/targets/aws_s3/tasks/main.yml]: The bucket is never deleted after this test, it looks like.

Comment (s-hertel) [reply test/integration/targets/aws_s3/tasks/main.yml]: Oops.

File: test/integration/targets/aws_s3/tasks/main.yml (hunk 1 above)
Comment (s-hertel) [reply test/integration/targets/aws_s3/tasks/main.yml]: But then it doesn't work on os x...
```
$ dd if=/dev/zero of=/tmp/test.txt bs=1M count=4096
dd: bs: illegal numeric value
```

**Comment by s-hertel** [comment]
I made the multipart test platform-specific because I couldn't find a non-gross alternative.

**Review by ryansb — APPROVED** [review]

**Pull request merged**